### ISSUE TITLE
newline_style = "Unix" has no effect for unchanged files

### ISSUE THREAD

**Issue by jstarks** [rust-lang/rustfmt#3939]
I have a project where many files have Windows line endings. I set newline_style to Unix and ran cargo fmt, expecting the files to be reformatted, but they are not unless there is some other non-line-ending change being made.

Interestingly, the opposite is not true: if I explicitly set newline_style to Windows, then any Unix-formatted files are rewritten even if there is no other reformatting to do.

To repro this:
```
cargo init
git add .
cargo fmt -- --config newline_style=Windows
git diff # see that src/main.rs has been reformatted with Windows line endings
git add -u
cargo fmt -- --config newline_style=Unix
git diff # see that there are no changes--src/main.rs still has Windows line endings

**Comment by calebcartwright** [comment]
What version of rustfmt are you using?

**Comment by jstarks** [comment]
Oops, sorry. This repros with both:
rustfmt 1.4.8-stable (afb1ee1c 2019-09-08)
and:
rustfmt 1.4.9-nightly (33e36670 2019-10-07)

**Comment by calebcartwright** [comment]
Thanks @jstarks - believe this is fixed in 1.4.10 via #3850

**Comment by jstarks** [comment]
Ah ha, you're right, confirmed. Closing. Thanks!
